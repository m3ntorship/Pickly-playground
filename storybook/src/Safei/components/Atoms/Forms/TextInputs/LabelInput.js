import React from "react";
import CorrectIcon from "../icons/Correct";
import WrongIcon from "../icons/Wrong";
import ArrowDown from "../icons/ArrowDown";

const LabelInput = ({ status, placeholder }) => {
  let statusStyle, iconDisplay;
  if (status === "error") {
    statusStyle = "border-error border";
    iconDisplay = <WrongIcon />;
  } else if (status === "success") {
    statusStyle = "border-success border";
    iconDisplay = <CorrectIcon />;
  } else {
    statusStyle = "border-dark border border-opacity-25";
    iconDisplay = "";
  }
  return (
    <>
      <div
        className={`${statusStyle} rounded-md  w-64 flex flex-row items-start content-center px-1 hover:border-grey-shd1 focus-within:border-opacity-100 bg-white label-container`}
      >
        {/* Enter Text */}
        <input
          placeholder={placeholder}
          style={{ width: "100%" }}
          className="px-3 py-1 outline-none"
          type="search"
        ></input>
        <div className="pt-2 pr-1 ">{iconDisplay}</div>
        <div className="pr-1 pt-2 cursor-pointer">
          <ArrowDown />
        </div>
      </div>
    </>
  );
};
export default LabelInput;
