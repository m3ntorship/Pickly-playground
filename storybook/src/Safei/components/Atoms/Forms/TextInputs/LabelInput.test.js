import React from "react";
import renderer from "react-test-renderer";
import LabelInput from "./LabelInput";

describe("my beverage", () => {
  it("renders when an input based on status ", () => {
    const tree = renderer.create(<LabelInput status="normal" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
