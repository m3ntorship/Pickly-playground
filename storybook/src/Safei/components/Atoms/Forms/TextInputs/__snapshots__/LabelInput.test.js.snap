// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Input field renders a marked input field with icon or prefix  1`] = `
<div
  className="border-dark border border-opacity-25 rounded-md  w-64 flex flex-row items-start content-center px-1 hover:border-grey-shd1 focus-within:border-opacity-100 bg-white label-container"
>
  <input
    className="px-3 py-1 outline-none"
    style={
      Object {
        "width": "100%",
      }
    }
    type="search"
  />
  <div
    className="pt-2 pr-1 "
  >
    
  </div>
  <div
    className="pr-1 pt-2 cursor-pointer"
  >
    <svg
      className="w-3 h-3"
      fill="#8D9A9E"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
        fillRule="evenodd"
      />
    </svg>
  </div>
</div>
`;
